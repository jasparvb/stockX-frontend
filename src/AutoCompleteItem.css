.AutoCompleteItem.active {
  background: rgb(0 200 5 / 5%);
}