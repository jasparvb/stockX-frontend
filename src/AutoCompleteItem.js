import React from "react";
import './AutoCompleteItem.css';

//Element displayed in search auto-suggest dropdown

function AutoCompleteItem({id, ticker, name, assetType, countryCode, selected, setSelected}) {
    return (
      <div 
        className={`AutoCompleteItem ${selected - 1 === id ? "active" : ""}`}
        onMouseEnter={() => setSelected(id + 1)}
      >
        <h3>{ticker} &ndash; {name}</h3>
        <p>{countryCode} | {assetType}</p>
      </div>
    );
}

export default AutoCompleteItem;