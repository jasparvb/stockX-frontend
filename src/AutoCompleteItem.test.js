import React from "react";
import { render } from "@testing-library/react";
import AutoCompleteItem from "./AutoCompleteItem";

it("renders without crashing", function() {
  render(<AutoCompleteItem />);
});

it("matches snapshot", function() {
  const { asFragment } = render(<AutoCompleteItem />);
  expect(asFragment()).toMatchSnapshot();
});
