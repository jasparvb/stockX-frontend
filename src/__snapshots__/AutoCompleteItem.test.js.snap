// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="AutoCompleteItem "
  >
    <h3>
       â€“ 
    </h3>
    <p>
       | 
    </p>
  </div>
</DocumentFragment>
`;
